src/components/ListItem.jsx

```diff
@@ -1,70 +1,45 @@
-import { useEffect, useState } from 'react';
+import { useState } from 'react';
 import './ListItem.css';
 import { updateItem } from '../api';
-import { handleNextPurchaseDate, ONE_DAY_IN_MILLISECONDS } from '../utils';
-import { increment } from 'firebase/firestore';
+import { calculateDateNextPurchased, ONE_DAY_IN_MILLISECONDS } from '../utils';
 
-// move current date outside of component to use it thoroughout the component
 const currentDate = new Date();
 
-export function ListItem({ item, listPath }) {
-	const [isPurchased, setIsPurchased] = useState(false);
-	// ---------------------------------------------------------------
-	// the path from the hook is glitching in case the user has multiple empty lists:
-	// navigates to the first empty list instead of the path that was clicked on
-	// const [listPath] = useStateWithStorage('tcl-shopping-list-path', null);
-	// ---------------------------------------------------------------
-	const {
-		name,
-		id,
-		dateLastPurchased,
-		dateCreated,
-		dateNextPurchased,
-		totalPurchases,
-	} = item;
-
-	useEffect(() => {
-		if (!dateLastPurchased) {
-			setIsPurchased(false);
-			return;
-		}
+const calculateIsPurchased = (dateLastPurchased) => {
+	if (!dateLastPurchased) {
+		return false;
+	} else {
```

Comment:
You can remove this `else` because the previous if conditional already returns early, so the code below the return won't be executed.

Reply:
makes sense! thank you